src/llmcompressor/transformers/finetune/data/base.py

```diff
@@ -106,142 +169,144 @@ def get_raw_dataset(self, cache_dir: Optional[str] = None) -> Dataset:
 
         return get_raw_dataset(
             self.data_args,
-            cache_dir,
+            None,
             split=self.split,
             streaming=self.data_args.streaming,
-            **self.raw_kwargs,
+            **self.data_args.raw_kwargs,
         )
 
-    def tokenize_and_process(
-        self, raw_dataset: Optional[Dataset] = None, add_labels: Optional[bool] = True
-    ) -> Dataset:
+    @cached_property
+    def preprocess(self) -> Union[Callable[[LazyRow], Any], None]:
```

Comment:
Both use cases are supported (passing in or applying manually). This is true now as it was previously as well.

Reply:
If the user passes a _preprocessed dataset_, then the `preprocess` step will be skipped since
1. Passing a custom `Dataset` object precludes using predefined datasets
2. The user will likely not pass a custom `preprocessing_func`. If they do, then the dataset gets preprocessed twice